### ISSUE TITLE
Make validation normative

### ISSUE THREAD

**Issue by jandrieu** [w3c/vc-data-model#1328]
For historical reasons, largely related to a perception that all normative requirements must be testable, there was push back against the language in #1297 that clarified that validation is normative.

IMO, this is a mistake.

Validation MUST occur before before verifiers rely on any claims.

This is, IMO, the simplest fact of validation from the earliest days of distinguishing verification from validation.

If validation is not required, then verifiers would be free to rely on credentials that have been verified but not validated.

However, it is exceptionally clear throughout our history that verifiers MUST trust the issuer for any claims they rely on. Checking to see if they trust a given issuer for a given claim is validation.

QED, validation is a normative requirement.

We should be clear about the requirement to validatate rather than obscuring it simply because it is untestable. It is untestable because the means of validation is out of scope. However, the act of validation is clearly required for the 3-party model to work at all.

**Comment by David-Chadwick** [comment]
`Validation MUST occur before before verifiers rely on any claims.`
Maybe the problem is that our model does not include the Relying Party, which is the entity that relies on the claims.
As I see it, verification by the verifier is an application independent function that is performed in the vc middle layer by the verifier.
Validation OTOH is an application dependent function and is performed by the RP in the application layer, and depends upon the business rules.
It is for this reason that many RPs could use the same third party provided verifier cloud service, since verification is a deterministic process that is independent of the business rules.

**Comment by jandrieu** [comment]
> `Validation MUST occur before before verifiers rely on any claims.` Maybe the problem is that our model does not include the Relying Party, which is the entity that relies on the claims. 

It definitely makes it hard to clarify the difference between the verifier as software, the verifier as a service,  and the verifier as the role. All of these have some expression in the VC-API work, but we've never bubbled up the distinctions to the VCDM.

FWIW, to my sense, the Verifier in VCs *is* the RP. They are the party that depends on the outcome of the verification process. But, I expect its too late to split the Verifier into the role of doing verification and the role of relying on the outcome.

> As I see it, verification by the verifier is an application independent function that is performed in the vc middle layer by the verifier. Validation OTOH is an application dependent function and is performed by the RP in the application layer, and depends upon the business rules. It is for this reason that many RPs could use the same third party provided verifier cloud service, since verification is a deterministic process that is independent of the business rules.

Exactly right. I see it similarly. The verification step is owned by the verifier/RP, but ultimately is executed by a pieces of software, possibly running on another entity's infrastructure.

IMO, the conflation between verification and validation are a direct result of our not having teased out both of these normative steps in the lifecycle of a VC. The facts have always been part of the larger story ("don't trust a issuer you don't know"), but we haven't spoken about it with the same clarity as most of our implementations do.

**Comment by msporny** [comment]
> Validation MUST occur before before verifiers rely on any claims.

Why can't we just say: Any entity that depends on any claims in a verifiable credential is expected to validate those claims before using them?

Why the push for normative RFC language? What does that language do that the language above doesn't accomplish?

Also note PR #1336 -- it feels like that PR could be related to this issue. I had some text in there that said that "verifiers SHOULD validate", but then removed it because it felt weird to say that verifiers also do validation (even though, I expect that's what we mean)? Is there a change to #1336 that would address this issue, @jandrieu?

**Comment by TallTed** [comment]
>> [@jandrieu] Validation MUST occur before before verifiers rely on any claims.
>
> [@msporny] Why can't we just say: Any entity that depends on any claims in a verifiable credential is expected to validate those claims before using them?

Either of the above would be a mistake. Once again, these would embed business logic — ours, not even the verifier's! — in the data model, which we've said many times goes beyond our remit.

It is up to the verifier to determine whether they rely on statements (claims) found in VCs, and if not, why not.

**Comment by dlongley** [comment]
Perhaps we can come up with some text that just makes a factual statement vs. rather than text that instructs verifiers on what to do / what they are expected to do.

@jandrieu said: "Checking to see if they trust a given issuer for a given claim is validation."

Could we say something like: "The process of verification only ensures that the claims in a verifiable credential are an authentic and timely statement made by the issuer. As anyone can issue a verifiable credential, any claims made by an untrusted issuer are not dissimilar to claims made by anyone at all."

**Comment by jandrieu** [comment]
@msporny has shifted my perspective on this, but for reasons that surprised me, so I want to share here.

We generally don't have any normative requirements required for Verifiers, but rather for verifier software. I reviewed the spec to see how that holds, and there was just one minor "should" in a non-normative section. We should review the spec to see that we have the same discipline for Issuers and Holders.

In other words, the spec doesn't normatively tell Roles what to do, but rather defines what normative software needs to do.

That's a clear and good distinction for me.

**Comment by msporny** [comment]
> @msporny has shifted my perspective on this, but for reasons that surprised me, so I want to share here.

*\*lol*\* ... and here I was thinking: "You know, Joe might be right here, we could probably define the algorithms for validating each field." <-- now, that would be A LOT of work at this point (right before we're going into CR), but Jeffrey's request for an algorithm for verification had me thinking about what it would take to define algorithms to validate each field. It's do-able... but probably 3-6 months of work (presuming we don't have any strong opinions on each of the algorithms, which we would).

> That's a clear and good distinction for me.

That's great -- if we could leave it there for now, I think the ecosystem will be ok for long enough for us to come back around to this in v3.0.

> I'll review the spec with that in mind.

Thank you! :)

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-11-28-vcwg-special#section1-1) on 2023-11-28

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.1. Make validation normative (issue vc-data-model#1328)

_See github issue [vc-data-model#1328](https://github.com/w3c/vc-data-model/issues/1328)._




**Manu Sporny:** This would be a lot of work.  
… There might be a way to do this in the future.  
… This could take months to resolve.  
… Joe is going to do a review.  
… Maybe validation doesn't need to be made normative.  

</details>

**Comment by msporny** [comment]
PR #1369 failed, so we're off the slippery slope.

**Comment by jandrieu** [comment]
I'm good to close. We can handle the language changes (if any) in the verification section.
